src/neo/Oracle/OraclePolicyContract.cs

```diff
@@ -0,0 +1,186 @@
+        }
+
+        public ECPoint[] GetOracleValidators(StoreView snapshot)
+        {
+            ECPoint[] consensusPublicKey = PolicyContract.NEO.GetValidators(snapshot);
+            IEnumerable<(ECPoint ConsensusPublicKey, ECPoint OraclePublicKey)> hasDelegateOracleValidators = GetDelegateOracleValidators(snapshot).Where(p => consensusPublicKey.Contains(p.ConsensusPublicKey));
+            hasDelegateOracleValidators.ToList().ForEach(p =>
+            {
+                var index = System.Array.IndexOf(consensusPublicKey, p.ConsensusPublicKey);
+                if (index >= 0) consensusPublicKey[index] = p.OraclePublicKey;
+            });
+            return consensusPublicKey;
+        }
+
+        public BigInteger GetOracleValidatorsCount(StoreView snapshot)
+        {
+            return GetOracleValidators(snapshot).Length;
+        }
+
+        internal IEnumerable<(ECPoint ConsensusPublicKey, ECPoint OraclePublicKey)> GetDelegateOracleValidators(StoreView snapshot)
+        {
+            byte[] prefix_key = StorageKey.CreateSearchPrefix(Id, new[] { Prefix_Validator });
+            return snapshot.Storages.Find(prefix_key).Select(p =>
+            (
+                p.Key.Key.AsSerializable<ECPoint>(1),
+                p.Value.Value.AsSerializable<ECPoint>(1)
+            ));
+        }
+
+        [ContractMethod(0_03000000, ContractParameterType.Boolean, ParameterTypes = new[] { ContractParameterType.Integer }, ParameterNames = new[] { "fee" })]
+        private StackItem SetTimeOutMilliSeconds(ApplicationEngine engine, Array args)
+        {
+            StoreView snapshot = engine.Snapshot;
+            Transaction tx = (Transaction)engine.ScriptContainer;
+            Array signatures = new Array(tx.Witnesses.ToList().Select(p => StackItem.FromInterface(p.VerificationScript)));
+            Array pubkeys = new Array(GetOracleValidators(snapshot).ToList().Select(p => StackItem.FromInterface(p)));
+            engine.CurrentContext.EvaluationStack.Push(signatures);
+            engine.CurrentContext.EvaluationStack.Push(pubkeys);
+            engine.CurrentContext.EvaluationStack.Push(StackItem.Null);
+
+            if (BitConverter.ToInt32(args[0].GetSpan()) <= 0) return false;
+            int timeOutMilliSeconds = BitConverter.ToInt32(args[0].GetSpan());
+            if (!InteropService.Crypto.ECDsaCheckMultiSig.Handler.Invoke(engine))
+            {
+                return false;
+            }
+            StorageKey key = CreateStorageKey(Prefix_TimeOutMilliSeconds);
+            if (snapshot.Storages.TryGet(key) != null)
+            {
+                StorageItem value = snapshot.Storages.GetAndChange(key);
+                value.Value = BitConverter.GetBytes(timeOutMilliSeconds);
+                return true;
+            }
+            else
+            {
+                snapshot.Storages.Add(key, new StorageItem
+                {
+                    Value = BitConverter.GetBytes(timeOutMilliSeconds)
```

Comment:
If we set in `Initialize`  method, we can remove this `else` branch.

Reply:
Got it.